packages/components/src/client/components/Tag.ts

```diff
@@ -0,0 +1,78 @@
+// Developer beware: this file is symlinked between components/ and theme/
+
+import { defineComponent, h,  } from "vue";
+import { useRoute, useRouter } from "vue-router";
+
+import type { VNode } from "vue";
+
+import "../styles/tag.scss";
+
+// TODO: Right now the magic number 9 is baked in
+// That's because access to the theme config is
+// tricky in non-theme components.
```

Comment:
The magic number does be tricky here.

But if you use the component lib in a third party theme, you are still be able to set it in `.vuepress/styles/hope-config.scss`. It's not a theme config.

Reply:
I just cannot came up an idea that how can script know the number of colors users are setting

Reply:
I was wondering.

Could you add some config to sass-palette:

~~~ ts
useSassPalette(app, {
  id: "abc",
  defaultConfig: "vuepress-plugin-abc/styles/config.scss",

 // +++
  sharedParams: [                
    "vuepress-plugin-abs/styles/tags.ts"
  ]

});
~~~

Then, we could have a TS config file

~~~ ts
const color_count = 12
const light = [], dark = []

// assume we have a color library somewhere...

const base_light = rgb("#ddccbb")
const base_dark = rgb("#887766")

for (let i = 0; i <color_count; i++) {
   light.push(color_rotate(base_light, i * 27
   dark.push(color_rotate(base_dark, i*27)
}

export default {
  color_count,
  light, 
  dark,
]
~~~

sass-palette would then take the maps passed to it in `shared_params` and use it to generate sass variables (in I guess config or palette).

Then... we set up a vite alias so that @styles points to «base»/styles, and then the components could access this configuration as

~~~ ts
import { color_count } from "@Styles/tags.ts"
~~~